components/gripper/trossen/trossen.go

```diff
@@ -81,162 +76,71 @@ func init() {
 		}, &AttrConfig{})
 }
 
-var (
-	portMapping   = map[string]*sync.Mutex{}
-	portMappingMu sync.Mutex
-)
-
-func getPortMutex(port string) *sync.Mutex {
-	portMappingMu.Lock()
-	defer portMappingMu.Unlock()
-	mu, ok := portMapping[port]
-	if !ok {
-		mu = &sync.Mutex{}
-		portMapping[port] = mu
-	}
-	return mu
-}
-
-// Gripper TODO.
+// Gripper represents an instance of a Trossen gripper.
 type Gripper struct {
-	jServo   *servo.Servo
-	moveLock *sync.Mutex
-	opMgr    operation.SingleOperationManager
+	name       string
+	trossenArm arm.LocalArm
+	logger     golog.Logger
 	generic.Unimplemented
 }
 
 // newGripper TODO.
-func newGripper(attributes *AttrConfig, logger golog.Logger) (gripper.LocalGripper, error) {
-	usbPort := attributes.SerialPath
-
-	jServo, err := findServo(usbPort, attributes.BaudRate, logger)
-	if err != nil {
-		return nil, err
-	}
-	if err := jServo.SetTorqueEnable(true); err != nil {
-		return nil, err
+func newGripper(name string, logger golog.Logger, deps registry.Dependencies) (gripper.LocalGripper, error) {
+	var _arm arm.LocalArm
+	// TODO: an arm name should be specified for the gripper as a configuration
+	// attribute in a future commit. This is a breaking change that needs to be
+	// scoped - GV
+	for _, d := range deps {
+		a, ok := d.(arm.LocalArm)
```

Comment:
[minor] is it possible to check that the arm is a trossen arm? This driver doesn't work otherwise right?

Reply:
Why would it go away if explicit? Wouldn't he need to check that it's a trossen model in Validate anyway? 

It's the type that I want to check since it doesn't make sense to mount this gripper on a non-trossen arm architecture.

Reply:
I meant it goes away because you don't have to search the dependency list if the arm is specified as an attribute, so the issue of accidentally grabbing the wrong arm during a search goes away. You know exactly which dep you're looking for and just use it.

I wasn't thinking of validation, because we're already past that stage when this code is being called, so it's already too late to actually return a config validation error. And even if you wanted to check it in Validate(), you can't because dependencies aren't actually ready at that stage, so the component can only validate based on fields in its own config.

Reply:
I'm not sure I follow. I was just thinking that you'd add a new attribute to the config here called "Arm" and it would be a string that named the specific arm. Then you'd just `trossenArm := deps[config.Arm].(generic.Generic)` and call DoCommand() against it. (Well, check the assertion is ok first.) I'm not sure what you're considering that needs private method/member access.

Reply:
I'm not sure where this thread ended, but +1 to James' comment. The arm should be an explicit config attribute rather than being implicitly deduced from the dependency list. This would eliminate the "multiple arms" case below. And would make it clearer for the user to configure.

Reply:
am I ok to do this? It's technically a breaking change